packages/beacon-node/src/execution/engine/http.ts

```diff
@@ -139,9 +158,19 @@ export class ExecutionEngineHttp implements IExecutionEngine {
       this.updateEngineState(getExecutionEngineState({payloadError: error, oldState: this.state}));
     });
 
-    this.rpc.emitter.on(JsonRpcHttpClientEvent.RESPONSE, () => {
-      this.updateEngineState(getExecutionEngineState({targetState: ExecutionEngineState.ONLINE, oldState: this.state}));
-    });
+    if (!opts?.disableClientVersionFetch) {
```

Comment:
This completely disabled engine state updates as well as it currently is but I would rather question the need for such an option altogether, if possible we should avoid options for testing only.

Can you please elaborate on why this is needed?

Reply:
@nflaig Reason why I need this is to pass `executionEngine/httpRetry.test.ts`, `executionEngine/http.test.ts`. Because the way they mock the server response will make every method returning the same expected response. This assumes each single test case will only trigger one rpc call but that's no longer the case because of `engine_getClientVersion`. The mock server will try to respond `engine_getClientVersionV1` with mocked payload/fcu responses but that will cause error because response format mismatch

Reply:
The way the tests work is not ideal but we should prefer to modify tests instead here imo instead of prod code